explainer.md

```diff
@@ -0,0 +1,255 @@
+```
+::shared-old(#root), ::shared-container(#shared-header) {
+  position: fixed;
+  top: 0px;
+  left: 0px;
+}
+
+::shared-container(#shared-header) {
+  // Container sized to the element's border-box size.
+  width: 100px;
+  height: 100px;
+  
+  // A transform mapping the element to its quad in viewport space.
+  transform: translate(0px, 0px);
+}
+
+::shared-old(#shared-header) {
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  content: element(#header);
+}
+
+::shared-old(#root) {
+  width: 100vw;
+  height: 100vh;
+  // The output of element() function on the root element.
+  content: element(html);
+}
+```
+
+3. Save the output of the element() function for each pseudo element referenced above with the computed size and transform applied to container elements. Then navigate to the new page leaving the last rendered pixels of the old page on screen.
+
+4. When the new page loads, suppress rendering until resources required for first render have been fetched. This state is currently driven by browser heuristics but the eventual goal is to give deterministic control to developers using a [renderblocking](https://github.com/whatwg/html/issues/7131) attribute.
+
+5. Once the page is ready for first render, create the following pseudo elements in the top layer. This is done until the end of rAF callbacks on the page's first rendering lifecycle update[^1] :
+    a. A container and child replaced element for each shared element on the old page using state saved in step 3.
+    b. A container and child replaced element for each element with shared-id attribute on the new page. Only the child element is created if a container with the matching id was already created in step a. above.
+    c. A replaced element for the root/html element identified via ::shared-old(#root).
+
+6. Apply the following UA stylesheet to the pseudo elements on the new page.
+```
+::shared-old(#root), ::shared-new(#root), ::shared-container(#shared-header) {
+  position: fixed;
+  top: 0px;
+  left: 0px;
+}
+
+::shared-old(#root), ::shared-new(#root) {
+  width: 100vw;
+  height: 100vh;
+  content: element(html);
+}
+::shared-old(#root) {
+  // This is the saved output referenced in step 3.
+  content: cached-element(html);
+}
+::shared-new(#root) {
+  content: element(html);
+}
+
+::shared-old(#shared-header), ::shared-new(#shared-header) {
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  
+  content: element(#header);
+}
+::shared-old(#shared-header) {
+  // This is the saved output referenced in step 3.
+  content: cached-element(#header);
+}
+::shared-new(#shared-header) {
+  content: element(#header);
+}
+
+// Update the container's size and transform to the shared element on the new page.
+::shared-container(#shared-header) {
+  width: 200px;
+  height: 200px;
+  transform: translate(0px, 100px);
+}
+```
+
+7. Once no pseudo element has an active animation, remove them from the top layer.
```

Comment:
Does this preclude the effect of "element doesn't move while the page underneath it shifts". Or is this an all or nothing?

Reply:
It's all or nothing. As long as any element has an animation, all top layer elements are alive. I'm definitely open to better alternatives here. Was planning to file an issue to discuss this and update the explainer.